runtime/jcl/common/sun_misc_Unsafe.cpp

```diff
@@ -100,11 +100,10 @@ Java_sun_misc_Unsafe_defineAnonymousClass(JNIEnv *env, jobject receiver, jclass
 
 	jsize length = env->GetArrayLength(bytecodes);
 
-	if (NULL != vm->sharedClassConfig) {
-		if (J9_ARE_ALL_BITS_SET(vm->sharedClassConfig->runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_SHAREANONYMOUSCLASSES)) {
-			/* if ShareAnonymousClasses is enabled, anonymous classes are shared and for isROMClassShareable() to return true, we need to set the flag of the classLoader */
-			vm->anonClassLoader->flags |= J9CLASSLOADER_SHARED_CLASSES_ENABLED;
-		}
+	if ((NULL != vm->sharedClassConfig) && J9_ARE_NO_BITS_SET(vm->sharedClassConfig->runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_SHAREANONYMOUSCLASSES)) {
```

Comment:
I don't think these assertions are required as they are checked in `isUnsafeClassSharingEnabled`

Reply:
These are for the anon class loader flag so they are not checked in `isUnsafeClassSharingEnabled` but I can still remove them if they are not necessary

Reply:
I think we're OK not asserting here as the SCC code is in a better position to validate the anon loader is SCC enabled or not.